File: app/src/main/java/com/veyndan/redditclient/ui/recyclerview/Swipeable.java
package com.veyndan.redditclient.ui.recyclerview;

import android.support.v7.widget.RecyclerView;

/**
 * If a {@link RecyclerView} has a {@link SwipeItemTouchHelperCallback} attached, then items in
 * that {@link RecyclerView} which wish to be swipeable should have their
 * {@link RecyclerView.ViewHolder} implement this interface.
 *
 * @see SwipeItemTouchHelperCallback
 */
public interface Swipeable {

    /**
     * Return {@code true} if this post should be swipeable, else {@code false}.
     */
    boolean swipeable();

    /**
     * Called when the current {@link RecyclerView.ViewHolder} is swiped by the user.
     * @see android.support.v7.widget.helper.ItemTouchHelper.Callback#onSwiped(RecyclerView.ViewHolder, int)
     */
    void onSwipe();
}

File: app/src/main/java/com/veyndan/paper/reddit/ui/recyclerview/Swipeable.kt
package com.veyndan.paper.reddit.ui.recyclerview;

/**
 * If a {@link RecyclerView} has a {@link SwipeItemTouchHelperCallback} attached, then items in
 * that {@link RecyclerView} which wish to be swipeable should have their
 * {@link RecyclerView.ViewHolder} implement this interface.
 *
 * @see SwipeItemTouchHelperCallback
 */
interface Swipeable {

    /**
     * Return {@code true} if this post should be swipeable, else {@code false}.
     */
    fun swipeable(): Boolean

    /**
     * Called when the current {@link RecyclerView.ViewHolder} is swiped by the user.
     *
     * @see android.support.v7.widget.helper.ItemTouchHelper.Callback#onSwiped(RecyclerView.ViewHolder, int)
     */
    fun onSwipe()
}
